public/app/features/plugins/components/restrictedGrafanaApis/alerting/alertingSchemaApi.ts

```diff
@@ -0,0 +1,208 @@
+import { z } from 'zod';
+
+import { RuleFormType } from 'app/features/alerting/unified/types/rule-form';
+import { GrafanaAlertStateDecision } from 'app/types/unified-alerting-dto';
+
+// Schema for __expr__ type queries (reduce, threshold, etc.)
+export const exprQuerySchema = z.object({
+  refId: z.string().describe('Reference ID for the query, e.g., "B", "C", etc.'),
+  type: z.string().describe('Expression type, e.g., "reduce", "threshold", "math"'),
+  datasource: z.object({
+    uid: z.literal('__expr__').describe('Must be "__expr__" for expression queries'),
+    type: z.literal('__expr__').describe('Must be "__expr__" for expression queries'),
+  }),
+  conditions: z
+    .array(
+      z.object({
+        type: z.string().describe('Condition type, e.g., "query"'),
+        evaluator: z.object({
+          params: z.array(z.any()).describe('Parameters for the evaluator'),
+          type: z.string().describe('Evaluator type, e.g., "gt", "lt", "within_range", "outside_range"'),
+        }),
+        operator: z.object({
+          type: z.string().describe('Operator type, e.g., "and", "or"'),
+        }),
+        query: z.object({
+          params: z.array(z.string()).describe('Query parameters, typically the refId to evaluate'),
+        }),
+        reducer: z.object({
+          params: z.array(z.any()).describe('Parameters for the reducer'),
+          type: z.string().describe('Reducer type, e.g., "last", "avg", "sum", "count", "min", "max"'),
+        }),
+      })
+    )
+    .optional()
+    .describe('Conditions for the expression query'),
+  reducer: z.string().optional().describe('Reducer function, e.g., "last", "avg", "sum"'),
+  expression: z.string().optional().describe('Expression referencing other queries, e.g., "A"'),
+  math: z.string().optional().describe('Math expression for math type queries'),
+});
+
+// Schema for regular datasource queries
+export const alertingQuerySchema = z.object({
+  refId: z.string().describe('Reference ID for the query, e.g., "A", "B", etc.'),
+  queryType: z.string().optional().default('alerting').describe('Type of query (e.g., "alerting", "recording")'),
+  expression: z
+    .string()
+    .optional()
+    .default('')
+    .describe('Query expression to be executed. This can not include variables (e.g. $var).'),
+  instant: z.boolean().optional().default(true).describe('Whether the query is an instant query'),
+  range: z
+    .boolean()
+    .optional()
+    .default(false)
+    .describe('Whether the query is a range query, should be false if instant is true'),
```

Comment:
I think these are Prometheus-specific fields

Reply:
good catch! i forgot removing these after moving back the cloud schema to our code.